### PR TITLE
enable warnings on simpll build & fix warnings

### PR DESCRIPTION
**Pull request by parallelistic**
PR for #202 

I'm unsure of what would be the best approach for -Wunused-parameters, I don't think spamming `__attribute__((unused))` everywhere in passes would be very nice to look at...
Other than that, I haven't performed any big changes to tests, nor do I think they would be needed.

SimpLL tests passed, and haven't run pytest, so if someone with the different kernels downloaded could do that, I would be grateful.

### CODE REVIEW

**Comment by viktormalik** [comment]
Thanks for the PR!

> I'm unsure of what would be the best approach for -Wunused-parameters, I don't think spamming `__attribute__((unused))` everywhere in passes would be very nice to look at... 

Making those arguments unnamed should be sufficient here.

> SimpLL tests passed, and haven't run pytest, so if someone with the different kernels downloaded could do that, I would be grateful.

CI took care of that ;-)

**Comment by parallelistic** [comment]
> > I'm unsure of what would be the best approach for -Wunused-parameters, I don't think spamming `__attribute__((unused))` everywhere in passes would be very nice to look at...
> 
> Making those arguments unnamed should be sufficient here.

Can't believe i didn't think of that 👍

File: diffkemp/simpll/CMakeLists.txt [hunk 1]
```diff
@@ -1,16 +1,20 @@
 # Create symlinks to a correct version of FunctionComparator based on LLVM version
-execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink
+execute_process(
+  COMMAND ${CMAKE_COMMAND} -E create_symlink
```
Comment (viktormalik) [inline diffkemp/simpll/CMakeLists.txt:3]: Please do not reformat lines that are not related to this PR, it makes `git blame` useless (this applies to almost the entire file).

File: diffkemp/simpll/DifferentialFunctionComparator.cpp [hunk 2]
```diff
@@ -600,7 +600,7 @@ bool DifferentialFunctionComparator::maySkipLoad(const LoadInst *Load) const {
                     searchPredecessors = false;
                     break;
                 }
-            } else if (auto Store = dyn_cast<StoreInst>(&*it)) {
+            } else if (/* auto Store = */ dyn_cast<StoreInst>(&*it)) {
```
Comment (viktormalik) [inline diffkemp/simpll/DifferentialFunctionComparator.cpp:603]:
```suggestion
            } else if (isa<StoreInst>(&*it)) {
```

File: diffkemp/simpll/DebugInfo.cpp [hunk 3]
```diff
@@ -148,8 +148,8 @@ void DebugInfo::extractAlignmentFromInstructions(GetElementPtrInst *GEP,
                 if (!TypeDISecond)
                     continue;
 
-                int indexSecond =
-                        getTypeMemberIndex(*TypeDISecond, elementName);
+                uint64_t indexSecond = static_cast<uint64_t>(
+                        getTypeMemberIndex(*TypeDISecond, elementName));
```
Comment (viktormalik) [inline diffkemp/simpll/DebugInfo.cpp:152]: Just use `(uint64_t)` here (and everywhere else). LLVM does the same, so we follow its approach.

**Review by viktormalik — CHANGES_REQUESTED** [review]

File: diffkemp/simpll/CMakeLists.txt (hunk 1 above)
Comment (parallelistic) [reply diffkemp/simpll/CMakeLists.txt:3]: My mistake, my vim autocmd was at fault and I realised as much before, which is made me run clang-format but then I forgot about the CMakeLists

File: diffkemp/simpll/CMakeLists.txt [hunk 4]
```diff
@@ -19,7 +23,7 @@ file(GLOB passes passes/*.cpp)
 file(GLOB library library/*.cpp)
 
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-rtti -fpic -fvisibility=hidden")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-rtti -fpic -fvisibility=hidden -Wall -Wextra -Wno-reorder")
```
[the remark this answers is not in the document]
Comment (parallelistic) [reply diffkemp/simpll/CMakeLists.txt:26]: It is in fact not, i had completely forgotten I had added it and therefor thought I finished fixing all errors :/

**Review by viktormalik — APPROVED** [review]
Looks good! Can you squash everything into a single commit which will describe a list of changes that were done?

**Comment by parallelistic** [comment]
> Looks good! Can you squash everything into a single commit which will describe a list of changes that were done?

Yup will do, was thinking about doing exactly that :)

**Comment by viktormalik** [comment]
Two last things related to the commit message:
1. please respect the 50/72 rule
2. if you don't mind, it'd be nice if you could use some actual (non-GitHub) email for your commits

**Comment by parallelistic** [comment]
done :+1:
